### ISSUE TITLE
Problems running with conda-forge, vscode, and mrsimulator

### ISSUE THREAD

**Issue by subhradip-paul** [deepanshs/mrsimulator#384]
Did you find a bug? Please help us improve by fill out the sections below. 👍

**Describe the bug**
The problem I am having while running mrsimulator is the following error:
*ModuleNotFoundError: No module named 'mrsimulator.base_model'*

I have installed miniforge, and mamba.
Then created a virtual environment called mrsim with the following dependencies:
pydantic < 2.0
numpy < 2
python = 3.9

**Screenshots**

*Code*
import os
os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE'] = '1' #setting the env variable
from mrsimulator import Simulator, SpinSystem, Site

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], [line 4](vscode-notebook-cell:?execution_count=1&line=4)
      [1](vscode-notebook-cell:?execution_count=1&line=1) import os
      [2](vscode-notebook-cell:?execution_count=1&line=2) os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE'] = '1' #setting the env variable
----> [4](vscode-notebook-cell:?execution_count=1&line=4) from mrsimulator import Simulator, SpinSystem, Site
      [5](vscode-notebook-cell:?execution_count=1&line=5) # from mrsimulator.method.lib import BlochDecaySpectrum
      [6](vscode-notebook-cell:?execution_count=1&line=6) # import matplotlib.pyplot as plt

File c:\Users\subhr\miniforge3\envs\mrsim\lib\site-packages\mrsimulator\__init__.py:69
     [66](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/__init__.py:66) os.environ["VECLIB_MAXIMUM_THREADS"] = "1"
     [67](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/__init__.py:67) # os.environ["NUMEXPR_NUM_THREADS"] = "1"
---> [69](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/__init__.py:69) from .spin_system import Site  # lgtm [py/import-own-module] # noqa:F401
     [70](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/__init__.py:70) from .spin_system import Coupling  # lgtm [py/import-own-module]  # noqa:F401
     [71](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/__init__.py:71) from .spin_system import SpinSystem  # lgtm [py/import-own-module] # noqa:F401

File c:\Users\subhr\miniforge3\envs\mrsim\lib\site-packages\mrsimulator\spin_system\__init__.py:9
      [6](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/spin_system/__init__.py:6) from typing import Union
      [8](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/spin_system/__init__.py:8) import numpy as np
----> [9](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/spin_system/__init__.py:9) from mrsimulator.base_model import get_zeeman_states
     [10](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/spin_system/__init__.py:10) from mrsimulator.transition import Transition
     [11](file:///C:/Users/subhr/miniforge3/envs/mrsim/lib/site-packages/mrsimulator/spin_system/__init__.py:11) from mrsimulator.transition.pathway import TransitionList

ModuleNotFoundError: No module named 'mrsimulator.base_model'
--------------------------------------------------------------------------------------------

**Desktop (please complete the following information):**

- OS: [Windows]
- Python version: [e.g. 3.9.19]
- `mrsimulator` version: [e.g. 0.7]

**Additional context**

Running it via vscode and selecting the virtual environment created.
On google colab things run fine.

**Comment by subhradip-paul** [comment]
If I have done something wrong in terms of installation, please let me know.
Thanks,
Paul

**Comment by deepanshs** [comment]
Hi Paul,

The problem appears to be missing DLL. The DLL is generated during the installation process.

May I ask how did you install mrsimulator inside the conda env? Was it with pip and if there was any error during the installation process?

-Deepansh

**Comment by subhradip-paul** [comment]
Hi Deepansh,

Thanks for your prompt response.

Generally I use mamba with conda, conda-forge channels. In order to use pip, I did the following steps:
1. mamba create -n mrsim
2. Activated the venv mrsim.
3. installed python 3.9 in that venv
4. installed pip, mamba install pip
5. and then pip install mrsimulator -u.

and, I have downgraded pydantic and numpy.

Did I do something wrong in terms of installation? If you suggest something else, I can try.

   Thanks,
Paul

**Comment by deepanshs** [comment]
Thanks for the details. I haven't tried mamba before. Let me try to reproduce the issue on my end and then get back to you. I assume there was no error during the `pip install mrsimulator -u` step.

**Comment by subhradip-paul** [comment]
No, there was no problem with the pip install mrsimulator -U step.

**Comment by deepanshs** [comment]
Hi Paul,

I tried the following steps to reproduce the issue on a Windows system, but all the steps were successful. I am still unsure of the cause of this issue.

**Steps used in an attempt to reproduce the error**
- Install miniforge for windows
- Open miniforge prompt, and from within the prompt, execute the following commands
  - `mamba create -n mrsim python=3.9`
  - `mamba activate mrsim`
  - `pip install mrsimulator`
  - `pip install "numpy<2" "pydantic<2" astropy`
  - `python test.py`

Here, test.py contains the 1D static and MAS example code from the mrsimulator GitHub readme doc.

You mentioned vscode. Are you running the python script from within vscode? Just so we are on the same page, can you run the python script from within the miniforge prompt and check if you still get the error?

-D

**Comment by subhradip-paul** [comment]
Hi Deepansh,

I created a new environment. It works fine on vscode also.
I think something went wrong in terms of creation of the virtual environment.

Everything works fine even from within vscode.

Thanks for the help, and sorry about posting an issue which is more related to the virtual environment that I created. I must have done something wrong with the environment.

I will close the issue.

Thanks again,
Paul

![image](https://github.com/user-attachments/assets/f9fbb5d4-1562-41d3-b820-dc70a0ac0838)
